### DISCUSSION TITLE
Change account

### DISCUSSION THREAD

**Discussion by radekskrabucha** [wevm/wagmi#1735] [Ideas]
I was trying to find feature to switch switch user's account that is currently connected. Unfortunately I didn't find it 😢 But I found the way how to do it and I'm wondering if maybe wagmi could support it, maybe in `useAccount` or `useConnect`?
Here is how to implement it: https://community.metamask.io/t/how-to-make-your-dapp-prompt-a-user-to-change-account-connection/2695

**Comment by allenchuang** [comment]
You can use the following 
```
  const { connector: activeConnector } = useAccount()

  useEffect(() => {
	const handleConnectorUpdate = ({account, chain}: ConnectorData) => {
      if (account) {
        console.log('new account', account)
      } else if (chain) {
        console.log('new chain', chain)
      }
    }

    if (activeConnector) {
      activeConnector.on('change', handleConnectorUpdate)
    }

	return () => activeConnector.off('change', handleConnectorUpdate)
  }, [activeConnector])
```
